### PR TITLE
Use filters for bitcoind backend syncing if available

### PR DESCRIPTION
**Pull request by benthecarman**
Built on top of #2916 just look at last commit

This makes it so we use filters instead of blocks to sync if it is available.

### CODE REVIEW

File: wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala [hunk 1]
```diff
@@ -345,8 +345,10 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
         Future.failed(new RuntimeException(
           s"Attempting to spend an ImmatureCoinbase ${out.outPoint.hex}, this should not be possible until it is confirmed."))
       case TxoState.ConfirmedSpent =>
-        Future.failed(new RuntimeException(
-          s"Attempted to mark an already spent utxo ${out.outPoint.hex} with a new spending tx ${spendingTxId.hex}"))
+        if (!out.spendingTxIdOpt.contains(spendingTxId)) {
```
Comment (Christewart) [inline wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala:348]: Hmm, did this happen to you in normal node operations? Why is this needed to be added here?

Comment (benthecarman) [reply wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala:348]: This happened when I moved back the sync height without deleting my db. Imo this should probably be kept incase we reprocess a tx that is confirmed.

File: app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala [hunk 2]
```diff
@@ -144,16 +144,17 @@ class BitcoinSServerMain(override val args: Array[String])
           tmpWallet)
         _ = logger.info("Starting wallet")
         _ <- wallet.start()
-        _ <- BitcoindRpcBackendUtil.syncWalletToBitcoind(bitcoind, wallet)
+        _ = BitcoindRpcBackendUtil
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala:159]: nit: 

Maybe 

```scala
start = System.currentTimeMillis
logger.info(s"Starting to sync wallet from bitcoind")
...
_ = logger.info(s"Done syncing wallet to bitcoind, took ${System.currentTimeMillis - start}ms")

**Review by Christewart — APPROVED** [review]
Took me ~10 minutes to sync 500,000 testnet blocks

**Comment by Christewart** [comment]
Does not compile

https://github.com/bitcoin-s/bitcoin-s/pull/2926/checks?check_run_id=2384860726#step:5:1036

**Review by Christewart — APPROVED** [review]
